include/deal.II/base/hdf5.h

```diff
@@ -1587,29 +1591,6 @@ namespace HDF5
 
 
 
-  template <>
-  inline bool
-  HDF5Object::get_attribute(const std::string &attr_name) const
```

Comment:
What;s the reason for getting rid of this specialization?

Reply:
This specialization is not necessary because if I added the bool support globally in `get_hdf5_datatype()`.

The bool support will be automatically added to `get_attribute()`, `set_attribute()`, `write_dataset()`, etc. That being said, I still have to test it!

Ideally we should get rid of all the specializations. But it is tricky to get rid of the `std_string` specialization because `std_string` is a UTF8 variable string data type. Therefore the HDF5 code is quite different.
```
template <>
inline std::string
  HDF5Object::get_attribute(const std::string &attr_name) const
```
I think that specializations are difficult to understand for somebody who didn't write the code, I think that C++20 `if constexpr` is cleaner.

How about I write two versions one with C++20 `if constexpr` and another one with the specialization? When deal.II will move to C++20 you can get rid of the old code. I'll use macros to separate the two versions.

Reply:
That's more code to maintain so if we need the specializations anyway, I prefer not to also have a separate version using `if constexpr`.

Reply:
Sounds good!